src/Honeycomb/Models/HoneycombApiSettings.cs

```diff
@@ -25,7 +27,19 @@ public string TeamId
         /// The default dataset to apply to each event.
         /// </summary>
         /// <value></value>
-        public string DefaultDataSet { get; set; }
+        public string DefaultDataSet
+        {
+            get
+            {
+                if (IsClassic) {
+                    return _dataset;
+                }
+                return string.IsNullOrWhiteSpace(_dataset) ?
+                    "unknown_service" :
+                    _dataset.Trim();
```

Comment:
warn on diff if we trimmed something (ala [go beeline](https://github.com/honeycombio/beeline-go/blob/dcd5cac17320d77d7242d157e4dea6c757edbd03/beeline.go#L146))